docs/v1/api.md

```diff
@@ -0,0 +1,230 @@
+# Service Catalog API
+
+This document proposes a Kubernetes-native service broker API. It is a formalization of [@krancour](http://github.com/krancour)'s [presentation](https://docs.google.com/presentation/d/1Fm3qG9zQ4R8ZbhtTObmQjO3UZYJ0ruYI4qr5w_gEKN8/edit?usp=sharing) that he gave at the sig-service-catalog meeting on 10/10/2016.
+
+# Motivations
+
+2 different demos were given at [10/3/2016’s sig-service-catalog meeting](https://www.youtube.com/watch?v=Kfar5Uw7CRg), both extremely promising. Each differed in quite a few ways, including how consumers used Kubernetes resources to communicate.
+
+This proposal aims to define a standard API, including a schema for Kubernetes-native resources, to:
+
+- Guide the continuing efforts shown in those demos
+- Improve the user experience (both cluster operator and application developer)
+- Attempt to extend the Kubernetes API instead of relying on comparatively clunky Third Party Resource types (long-term goal)
+- Ensure a consistent user experience regardless of implementation, without undermining competition or choice
+
+# Inspiration
+
+Broadly speaking, we're proposing a system that can list services and allow consumers to provision, bind, unbind and deprovision them.
+
+We'll draw inspiration from the [Persistent Volumes](http://kubernetes.io/docs/user-guide/persistent-volumes/) API and the [Ingress Controllers](http://kubernetes.io/docs/user-guide/ingress/) system. The remainder of this section explains these inspirations in further detail.
+
+## Persistent Volumes
+
+First, this problem space is symmetric to that of persistent volumes. We propose to model the data structures herein after those in persistent volumes primarily because we believe that consistency is a virtue. More importantly, the community benefits when similar problems have similar solutions.
+
+Specifically, we believe that the following existing types provide a good model for a service catalog API:
+
+- `StorageClass`
+- `PersistentVolume`
+- `PersistentVolumeClaim`
+
+## Ingress Controllers
+
+Today, operators choose their own implementation of an ingress controller. Kubernetes gives the operator the ability to specify any ingress controller, and the complete freedom to implement it themselves.
+
+Similarly, permitting operators to implement or select their own service catalog controller would encourage freedom of choice and diversity. We propose a system similar to that defined in ingress controllers to do so.
+
+# Proposed Resources
+
+Given these inspirations, we propose adding the following resources to the Kubernetes API:
+
+- `ServiceClass`
+  - Describes a _kind_ of service (similar to how a `StorageClass` defines a _kind_ of persistent volume)
```

Comment:
For context, I happen to have been involved with getting `StorageClass` conceptualized and proposed in Kubernetes, so I have some background in that area too.

The storage problem space is very slightly different from what you propose.  Storage classes are really parameter sets that a provisioner picks up, rather than kinds of volumes.  The concept of 'kind' to me here is more coarse-grained, like 'EBS vs. GCE PD'.  So, while I don't think you necessarily need to change any of the text here, I just want to call out that the storage problem space is not an exact mirror image of this one, but is definitely similar enough to provide some precedent and inspiration.

To be honest, I think service catalog can solve some of the problems we have left to solve in the storage area, but that's a digression I'll avoid making for now :)